tests/ci/stress.py

```diff
@@ -43,6 +43,10 @@ def get_options(i, upgrade_check):
             client_options.append("join_algorithm='auto'")
             client_options.append("max_rows_in_join=1000")
 
+    if i % 3 == 1:
```

Comment:
The same condition is used for `join_use_nulls=1`, so it will always test `join_use_nulls` with query cache and query cache with `join_use_nulls`. We use deterministic conditions for historical reasons mostly. Let's try `i > 0 and rand() % 3 == 0` for these settings (`i > 0` is to run the first thread with default settings)

Reply:
Sounds much better :-)